packages/block-library/src/comment-template/edit.js

```diff
@@ -23,11 +28,89 @@ const TEMPLATE = [
 	[ 'core/comment-edit-link' ],
 ];
 
-function CommentTemplateInnerBlocks() {
+function CommentTemplateInnerBlocks( {
+	comments,
+	activeBlockContext,
+	setActiveBlockContext,
+	blocks,
+} ) {
 	const innerBlocksProps = useInnerBlocksProps( {}, { template: TEMPLATE } );
-	return <li { ...innerBlocksProps } />;
+	return (
+		<li { ...innerBlocksProps }>
+			{ innerBlocksProps.children }
+			{ comments.length > 0 ? (
+				<RenderComments
+					blockContexts={ comments }
+					activeBlockContext={ activeBlockContext }
+					setActiveBlockContext={ setActiveBlockContext }
+					blocks={ blocks }
+				/>
+			) : null }
+		</li>
+	);
 }
 
+/**
+ * Component that renders a list of (nested) comments. It is called recursively
+ * in its own body.
+ *
+ * @param {Object} props                         Component props.
+ * @param {Array}  [props.blockProps]            - Props from parent's `useBlockProps()`.
+ * @param {Array}  [props.blockContexts]         - Array of comment objects.
+ * @param {Array}  [props.activeBlockContext]    - The block that is currently active.
+ * @param {Array}  [props.setActiveBlockContext] - The setter for activeBlockContext.
+ * @param {Array}  [props.blocks]                - Array of blocks returned from
+ *                                               getBlocks() in parent .
+ * @return {WPElement}                 		List of comments.
+ */
+const RenderComments = ( {
+	blockProps,
+	blockContexts,
+	activeBlockContext,
+	setActiveBlockContext,
+	blocks,
+} ) => (
+	<ol { ...blockProps }>
+		{ blockContexts &&
+			blockContexts.map( ( blockContext ) => (
+				<BlockContextProvider
+					key={ blockContext.commentId }
+					value={ blockContext }
+				>
+					{ blockContext ===
+					( activeBlockContext || blockContexts[ 0 ] ) ? (
+						<CommentTemplateInnerBlocks
+							comments={ blockContext.children }
+							activeBlockContext={ activeBlockContext }
+							setActiveBlockContext={ setActiveBlockContext }
+							blocks={ blocks }
+						/>
+					) : (
+						<li>
+							<BlockPreview
+								blocks={ blocks }
+								__experimentalLive
+								__experimentalOnClick={ () =>
+									setActiveBlockContext( blockContext )
+								}
+							/>
+							{ blockContext.children.length > 0 ? (
+								<RenderComments
+									blockContexts={ blockContext.children }
+									activeBlockContext={ activeBlockContext }
+									setActiveBlockContext={
+										setActiveBlockContext
+									}
+									blocks={ blocks }
+								/>
+							) : null }
+						</li>
```

Comment:
My reasoning was that we could also move all those lines to `CommentTemplateInnerBlocks`. If I read it correctly we would only have to branch the part which decides whether to render the `BlockPreview` component or `nnerBlocksProps.children`. Unless we should be concerned with the `useInnerBlocksProps` hook call that would be unnecessary when `BlockPreview` gets rendered.

Reply:
Oh, yeah, you're right, that's a little more readable then :)  I thought that this might be possible but I didn't actually try it.